pandas/core/dtypes/dtypes.py

```diff
@@ -931,10 +931,15 @@ def construct_from_string(cls, string):
         attempt to construct this type from a string, raise a TypeError
         if its not possible
         """
-        if (isinstance(string, compat.string_types) and
-            (string.startswith('interval') or
-             string.startswith('Interval'))):
-            return cls(string)
+        if isinstance(string, compat.string_types):
+            if (string.lower() == 'interval' or
+               cls._match.search(string) is not None):
+                    return cls(string)
+            else:
+                msg = ('Incorrectly formatted string passed to constructor. '
```

Comment:
you don't actually need the else here

Reply:
the else statement is now removed. thanks for the help! 

cases from the original post are covered in pandas/core/tests/dtypes/test_dtypes.py

pd.api.types.IntervalDtype('IntervalA') covered in line 514
pd.api.types.IntervalDtype.construct_from_string('IntervalA') is covered indirectly but got lost in the shuffle, I will re-add this one explicitly 
pd.api.types.IntervalDtype.is_dtype('IntervalA') covered in line 565

Reply:
pd.api.types.IntervalDtype.construct_from_string('IntervalA') now covered in line 536